testing/trino-product-tests/src/main/java/io/trino/tests/jdbc/TestExternalAuthorizerOAuth2.java

```diff
@@ -87,6 +88,28 @@ public void shouldAuthenticateAndExecuteQuery()
         }
     }
 
+    @Test(groups = {OAUTH2, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/6991", match = "Last login action has failed with exception")
+    public void shouldAuthenticateAfterTokenExpires()
+            throws SQLException, InterruptedException
+    {
+        prepareSeleniumHandler();
+        Properties properties = new Properties();
+        properties.setProperty("user", "test");
+        try (Connection connection = DriverManager.getConnection(jdbcUrl, properties);
+                PreparedStatement statement = connection.prepareStatement("select * from tpch.tiny.nation");
+                ResultSet results = statement.executeQuery()) {
+            assertThatNoException().isThrownBy(() -> checkLastLoginActionExecutionFailures());
+            SECONDS.sleep(5);
```

Comment:
sleep more than 5 seconds, add a comment why that long (mention `HydraIdentityProvider.TTL_ACCESS_TOKEN_IN_SECONDS`)